src/extension-runners/firefox-android.js

```diff
@@ -269,6 +273,21 @@ export class FirefoxAndroidExtensionRunner {
 
   // Private helper methods.
 
+  get fennecCompatibilityMode() {
+    if (this.params.fennecMode != null) {
+      return this.params.fennecMode;
+    }
+
+    const firefoxApk =
+      this.selectedFirefoxApk || this.params.firefoxApk;
+
+    if (firefoxApk && firefoxApk.includes('.fenix')) {
+      return false;
+    }
+
```

Comment:
> Is it possible to detect the application version before assuming Fenix? I.e. if <=68.x, assume Fennec, otherwise Fenix.

I looked into it a bit, but I haven't found a reasonable way to do that yet, `pm list packages` doesn't seem to provide version details for the listed packages.

Personally I would prefer to keep this pull request as simple as possible, so that we may even release it as part of the next minor version (which I haven't published yet).

Nevertheless I'm definitely interested on improving the Fenix support and auto-detection in followups.

Reply:
Fenix is a moving target, whereas we know what Fennec looks like. In the long term, we should be able to assume Fenix by default, unless.

This is not a blocker for this PR, so you can file a follow-up to address this.